test/client.test.js

```diff
@@ -401,7 +401,10 @@ describe('ServiceClient', () => {
         let numberOfRetries = 0;
         clientOptions.retryOptions = {
             retries: 3,
-            onRetry() {
+            onRetry(currentAttempt, err, req) {
+                assert(typeof currentAttempt === 'number');
+                assert(typeof err.type === 'string');
+                assert(req.pathname === '/');
```

Comment:
I think it's not very obvious that this test case fails when one of these assertions fails. It will fail implicitly by the assertion for `numberOfRetries` though.

How about using a sinon's stub for `onRetry` and explicitly assert its call count and arguments? Then we also don't need `numberOfRetries` variable.